okhttp/src/main/kotlin/okhttp3/internal/connection/ConnectPlan.kt

```diff
@@ -393,17 +398,17 @@ class ConnectPlan(
       // Success! Save the handshake and the ALPN protocol.
       val maybeProtocol =
         if (connectionSpec.supportsTlsExtensions) {
-          Platform.get().getSelectedProtocol(sslSocket)
+          Platform.get().getSelectedProtocol(sslSocket.delegate)
```

Comment:
Probably a TODO, but if you look at SocketAdapter, ConscryptSocketAdapter, Android10SocketAdapter, these are ideally something that would be hidden behind the OkioSslSocketFactory API, not through Platform.

Reply:
I think this socket adapter experience was why I kept suggestion an okio socket. This change looks like a good start.